store/mongo/datastore_mongo.go

```diff
@@ -361,37 +423,107 @@ func (db *DataStoreMongo) GetSessionRecording(ctx context.Context,
 		return err
 	}
```

Comment:
The cursor is never closed.
```suggestion
	}
	defer recordingsCursor.Close()
```

Reply:
good catch. the problem is: I cant close it in there, it is passed to `NewRecordingReader` and used in `(*RecordingReader).Read`, so instead I am closing it there, when there are no more records. I have to think how to prevent eventual leaking if the `(*RecordingReader).Read` is not called, currently have no idea how to do it.

Reply:
> good catch. the problem is: I cant close it in there, it is passed to `NewRecordingReader` and used in `(*RecordingReader).Read`, so instead I am closing it there, when there are no more records. I have to think how to prevent eventual leaking if the `(*RecordingReader).Read` is not called, currently have no idea how to do it.

no wait. I am an idiot. sorry. fixing.